tests/unit/test_stream_operations.py

```diff
@@ -0,0 +1,167 @@
+import time
```

Comment:
I'm quite against this entire test. We're testing some internal implementation instead of behavior.

What we should do instead is write a unit test for sync process mocking the repository and the backend client.